### PR TITLE
CAN-918

### PR DESCRIPTION
**Pull request by ks-rahul**
1. CAN-1022
2. CAN-1024

### CODE REVIEW

File: src/components/ComponentPages/CreateNewCamp/index.tsx [hunk 1]
```diff
@@ -87,6 +87,31 @@ const CreateNewCamp = ({
...
+      node = list[i];
+      if (node.parentId !== "0") {
+        // if you have dangling branches check that map[node.parentId] exists
+        list[map[node.parentId]]?.children.push(node);
+      } else {
+        roots.push(node);
+      }
+    }
+
+    console.log("[ROOT]", roots);
```
Comment (akuks) [inline src/components/ComponentPages/CreateNewCamp/index.tsx]: Remove console.logs

File: src/components/ComponentPages/CreateNewCamp/index.tsx [hunk 2]
```diff
@@ -87,6 +87,31 @@ const CreateNewCamp = ({
     }
   };
 
+  function list_to_tree(list) {
+    var map = {},
+      node,
+      roots = [],
+      i;
+
+    for (i = 0; i < list.length; i += 1) {
```
Comment (akuks) [inline src/components/ComponentPages/CreateNewCamp/index.tsx]: for is repeated. Can we merge both?

File: src/components/ComponentPages/CreateNewTopic/UI/FromUI.tsx [hunk 3]
```diff
@@ -37,12 +37,12 @@ const CreateTopicFromUI = ({
       <Card
         title={CardTitle}
         className="can-card-style"
-        extra={
-          <PreventSubCamps
-            options={options}
-            onCheckboxChange={onCheckboxChange}
-          />
-        }
+        // extra={
```
Comment (akuks) [inline src/components/ComponentPages/CreateNewTopic/UI/FromUI.tsx]: Remove the code if not required

File: src/components/ComponentPages/CreateNewTopic/index.tsx [hunk 4]
```diff
@@ -65,7 +65,7 @@ const CreateNewTopic = ({
       note: values.edit_summary?.trim(),
     };
 
-    options.map((op) => (body[op.id] = op.checked ? 1 : 0));
+    // options.map((op) => (body[op.id] = op.checked ? 1 : 0));
```
Comment (akuks) [inline src/components/ComponentPages/CreateNewTopic/index.tsx]: Delete the commented code

File: src/components/ComponentPages/Statement/AddOrManage/index.tsx [hunk 5]
```diff
@@ -198,7 +198,7 @@ export default function AddOrManage({ add }) {
     } else if (manageFormOf == "statement") {
       res = await updateStatementApi(reqBody);
     } else if (manageFormOf == "topic") {
-      options.map((op) => (reqBody[op.id] = op.checked ? 1 : 0));
+      // options.map((op) => (reqBody[op.id] = op.checked ? 1 : 0));
```
Comment (akuks) [inline src/components/ComponentPages/Statement/AddOrManage/index.tsx]: Delete the commented code

File: src/utils/generalUtility.ts [hunk 6]
```diff
@@ -277,3 +277,31 @@ export const showCreateCampButton = (camp: {
...
+}) => {
+  if (camp?.is_disabled === 1) {
+    if (camp?.parent_camp_is_one_level == undefined) {
+      // return true;
+    } else if (camp?.parent_camp_is_disabled == undefined) {
+      // return true;
+    }
+    return true;
+  } else if (camp?.is_disabled === 0) {
+    if (camp?.parent_camp_is_disabled === 1) {
```
Comment (akuks) [inline src/utils/generalUtility.ts]: Please refactor the nested ifs

**Review by akuks — CHANGES_REQUESTED** [review]

File: src/components/ComponentPages/CreateNewCamp/index.tsx (hunk 2 above)
Comment (ks-rahul) [reply src/components/ComponentPages/CreateNewCamp/index.tsx]: removed

File: src/components/ComponentPages/CreateNewCamp/index.tsx (hunk 1 above)
Comment (ks-rahul) [reply src/components/ComponentPages/CreateNewCamp/index.tsx]: removed

File: src/components/ComponentPages/CreateNewTopic/UI/FromUI.tsx (hunk 3 above)
Comment (ks-rahul) [reply src/components/ComponentPages/CreateNewTopic/UI/FromUI.tsx]: removed

File: src/components/ComponentPages/CreateNewTopic/index.tsx (hunk 4 above)
Comment (ks-rahul) [reply src/components/ComponentPages/CreateNewTopic/index.tsx]: removed

File: src/components/ComponentPages/Statement/AddOrManage/index.tsx (hunk 5 above)
Comment (ks-rahul) [reply src/components/ComponentPages/Statement/AddOrManage/index.tsx]: deleted

File: src/utils/generalUtility.ts (hunk 6 above)
Comment (ks-rahul) [reply src/utils/generalUtility.ts]: refactored

**Review by akuks — APPROVED** [review]

**Pull request merged**